GUI/test/javafxgui/controller/GUIAppControllerTest.java

```diff
@@ -107,27 +112,26 @@ public void handleReplayEvent() {
                 X, X, EMPTY
         };
 
-        GUIGameMakerFake gameMaker = new GUIGameMakerFake(new GUIPlayerFactory());
-        gameMaker.setDummyBoard(arrayToList(currentBoard));
-        Controller controller = new TTTController(gameViewSpy, gameMaker);
-        controller.startGame(GUI_HVH, 9);
+        fakeGameMaker.setDummyBoard(arrayToList(currentBoard));
+        GUIAppController controller = new GUIAppController(fakeGameMaker, gameViewSpy);
+        controller.startGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
         assertEquals(true, gameViewSpy.hasReplayButtonBeenDisplayed());
     }
 
     @Test
     public void clickingReplayButtonGeneratesANewGame() {
-        Controller controller = new TTTController(gameViewSpy, gameMaker);
+        GUIAppController controller = new GUIAppController(gameMaker, gameViewSpy);
         controller.startGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
         assertEquals(true, gameViewSpy.hasGameLayoutBeenRendered());
         assertEquals(false, gameViewSpy.hasGameOptionsBeenPresented());
-        controller.createNewGame();
+        controller.displayGameOptions();
         assertEquals(true, gameViewSpy.hasGameOptionsBeenPresented());
```

Comment:
the asserts before the "act" is a smell in here. i know that you're trying to make sure that hasGameOptionsBeenPresented() changes value, i'd make that more explicit perhaps. how could we do that?